pkg/apis/meta/v1alpha1/image.go

```diff
@@ -47,3 +45,89 @@ func (i *OCIImageRef) UnmarshalJSON(b []byte) error {
 	*i, err = NewOCIImageRef(str)
 	return err
 }
+
+func ParseOCIContentID(str string) (*OCIContentID, error) {
+	named, err := reference.ParseDockerRef(str)
+	if err != nil {
+		return nil, err
+	}
+
+	if canonical, ok := named.(reference.Canonical); ok {
+		return &OCIContentID{
+			repoName: named.Name(),
+			digest:   canonical.Digest().String(),
+		}, nil
+	}
+
+	d, err := digest.Parse(str)
+	if err != nil {
+		return nil, err
+	}
+
+	return &OCIContentID{
+		digest: d.String(),
+	}, nil
+}
+
+type OCIContentID struct {
+	repoName string // Fully qualified image name, e.g. "docker.io/library/node" or blank if the image is local
+	digest   string // Repo digest of the image, or sha256sum provided by the source if the image is local
+}
+
+var _ json.Marshaler = &OCIContentID{}
+var _ json.Unmarshaler = &OCIContentID{}
+
+func (o *OCIContentID) String() string {
+	if len(o.repoName) > 0 {
+		return fmt.Sprintf("oci://%s", o.RepoDigest())
+	}
+
+	return fmt.Sprintf("docker://%s", o.Digest())
+}
+
+func parseOCIString(s string) (*OCIContentID, error) {
+	u, err := url.Parse(s)
+	if err != nil {
+		return nil, err
+	}
+
+	// Remove the "docker://" or "oci://" scheme by only caring about the host and path
+	return ParseOCIContentID(u.Host + u.Path)
+}
+
+// Local returns true if the image has no repoName, i.e. it's not available from a registry
+func (o *OCIContentID) Local() bool {
+	return len(o.repoName) == 0
+}
+
+// Digest is a getter for the digest field
+func (o *OCIContentID) Digest() string {
```

Comment:
return `digest.Digest`? I think that would make sense

Reply:
Done in d4495d0 and also for `RepoDigest()`